proto/sequencerapis/astria/sequencer/v1alpha1/service.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package astria.sequencer.v1alpha1;
+
+import "astria/sequencer/v1alpha1/block.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+
+message GetSequencerBlockRequest {
+  // The height of the block to retrieve.
+  uint64 height = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message FilteredSequencerBlockRequest {
+  // The height of the block to retrieve.
+  uint64 height = 1 [(google.api.field_behavior) = REQUIRED];
+  // The 32 bytes identifying a rollup. Usually the sha256 hash of a plain rollup name.
+  repeated bytes rollup_ids = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+service SequencerService {
+  // Given a block height, returns the sequencer block at that height.
+  rpc GetSequencerBlock(GetSequencerBlockRequest) returns (SequencerBlock) {
+    option (google.api.http) = {get: "/v1alpha1/sequencer/{height}"};
+  }
+
+  // Given a block height and set of rollup ids, returns a SequencerBlock which
+  // is filtered to contain only the transactions that are relevant to the given rollup.
+  rpc FilteredSequencerBlock(FilteredSequencerBlockRequest) returns (SequencerBlock) {
```

Comment:
To be symmetric with the other functions, this should IMO called `GetFilteredSequencerBlock` (although I still prefer a more explicit naming like `GetSequencerBlockFilteredByRollupIds` or something like that).

Also, this should return a different value - name it `FilteredSequencerBlock` or `SequencerBlockSubset` or something like that. These objects have different semantics.

Reply:
updated